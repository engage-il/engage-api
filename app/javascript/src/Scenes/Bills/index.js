// @flow
export { Bills } from './Bills'
