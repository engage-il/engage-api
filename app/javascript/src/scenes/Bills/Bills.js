// @flow
import * as React from 'react'
import { withRouter } from 'react-router-dom'
import type { ContextRouter } from 'react-router-dom'
import { graphql } from 'react-relay'
import { RelayRenderer, BillSearch } from '@/components'
import type { Viewer } from '@/types'

const PAGE_SIZE = 20

type Props = {
  viewer: ?Viewer
} & ContextRouter

function paramsFromRoute ({ location }: ContextRouter) {
  return location.state && location.state.params
}

let Bills = function Bills ({ viewer, ...props }: Props) {
  return (
    <BillSearch
      viewer={viewer}
      /* $FlowFixMe: intersection & rest types */
      params={paramsFromRoute(props)}
      pageSize={PAGE_SIZE}
    />
  )
}

Bills = withRouter(Bills)

export { Bills }

export function BillsRenderer () {
  const query = graphql`
    query BillsQuery(
      $filter: BillsSearchFilter!,
      $count: Int!,
      $cursor: String!
    ) {
      viewer {
        ...BillSearch_viewer
      }
    }
  `

  return (
    <RelayRenderer
      // $FlowFixMe: make root accept class / function components after upgrading flow
      root={Bills}
      query={query}
      getVariables={(props) => ({
        count: PAGE_SIZE,
        cursor: '',
        filter: {
          key: 'bills',
          query: '',
          ...paramsFromRoute(props)
        }
      })}
    />
  )
}
