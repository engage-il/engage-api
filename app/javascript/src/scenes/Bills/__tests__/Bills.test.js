/* eslint-env jest */
import * as React from 'react'
import { shallow } from 'enzyme'
import { defaultsDeep } from 'lodash'
import { Bills } from '../Bills'

// subject
let subject

const defaults = {
  viewer: 'test-viewer',
  location: {
    state: null
  }
}

function loadSubject (props = {}) {
  subject = shallow(<Bills {...defaultsDeep(props, defaults)} />)
}

// specs
afterEach(() => {
  subject = null
})

describe('#render', () => {
  it('shows the home scene', () => {
    loadSubject()
    expect(subject).toMatchSnapshot()
  })

  it('shows the bill search with the correct params', () => {
    loadSubject({
      location: {
        state: {
          params: 'test-params'
        }
      }
    })

    expect(subject).toMatchSnapshot()
  })
})
