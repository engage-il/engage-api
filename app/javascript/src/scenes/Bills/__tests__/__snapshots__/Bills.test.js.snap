// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render shows the bill search with the correct params 1`] = `
<BillSearch
  pageSize={20}
  params="test-params"
  viewer="test-viewer"
/>
`;

exports[`#render shows the home scene 1`] = `
<BillSearch
  pageSize={20}
  params={null}
  viewer="test-viewer"
/>
`;
